src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -619,6 +619,13 @@ impl TargetInfo {
             .iter()
             .any(|sup| sup.as_str() == split.as_str())
     }
+
+    /// Checks if a target support core only.
+    ///
+    /// If no explictly stated in target spec json, we treat it as "maybe support".
+    pub fn support_core_only(&self) -> bool {
+        matches!(self.supports_std, Some(false))
```

Comment:
Unsure if this is something inherent to build-std or something weird about how we have the code structured but looking at this function in isolated, it seems weird to say that something only supports `core` if `std` isn't supported.  There is also `alloc` and others, right?

Also, outside of the context of buiild-std, this makes it read like we have native support for no_std crates but this is separate from that.

Reply:
Thanks for calling all these out.

Just updated the function name to `maybe_support_std` and called out that it is for `-Zbuild-std`.

> it seems weird to say that something only supports `core` if `std` isn't supported.  There is also `alloc` and others, right?

Yes, but core is like the fundamental of everything and `alloc` depends on `core`.

Anyway, this only affects `-Zbuild-std` to determine the default set of crate to build, if no value is provided to `-Zbuild-std`.